### PR TITLE
[fix]Replace dist.gather with dist.all_gather for hccl backend compatibility

### PR DESCRIPTION
**Pull request by 1329009851**
## Motivation

fix issue: https://github.com/sgl-project/sgl-kernel-npu/issues/669
Replace dist.gather with dist.all_gather for hccl backend compatibility

## Modifications

 Replace dist.gather calls with dist.all_gather in test_internode, test_internode_a2, and test_intranode so that all ranks allocate the gather buffer list, matching the workaround already applied in tests/python/deepep/utils.py.

## Testing

The test script is verified after the modification.

## Checklist

- [x] Format your code.
- [x] Add unit tests.
- [x] Update documentation.
- [x] Provide accuracy and speed benchmark results.

### CODE REVIEW

File: tests/python/deepep/test_internode.py [hunk 1]
```diff
@@ -415,12 +415,8 @@ def test_diagnose(
         ):
             if stats is None:
                 continue
-            gather_list = (
-                [torch.zeros_like(stats) for _ in range(group.size())]
-                if rank == 0
-                else None
-            )
-            dist.gather(stats, gather_list=gather_list, group=group, dst=0)
+            gather_list = [torch.zeros_like(stats) for _ in range(group.size())]
```
Comment (kaniel-outis) [inline tests/python/deepep/test_internode.py:418]: 使用issue跟踪一下。
